src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,106 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int currentCap;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        this.currentCap = DEFAULT_CAPACITY;
+        this.table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (LOAD_FACTOR * currentCap == size) {
+            resize();
+        }
+        Node<K, V> bucket = new Node<>(hash(key), key, value);
+        if (table[bucket.hash] == null) {
+            table[bucket.hash] = bucket;
+        } else {
+            Node<K, V> iterator = table[bucket.hash];
+            while (iterator != null) {
+                if (iterator.key == key || iterator.key != null
+                        && iterator.key.equals(key)) {
+                    iterator.value = value;
+                    return;
+                }
+                if (iterator.next == null) {
+                    break;
+                }
+                iterator = iterator.next;
+            }
+            iterator.next = bucket;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        if (size == 0) {
+            return null;
+        }
+        for (Node<K, V> bucket : table) {
```

Comment:
let's iterate over 1 bucket where lies the key (we can calculate bucket index from hashcode)